src/infrastructure/storage/wallet-repository.ts

```diff
@@ -0,0 +1,467 @@
+        await callback(txID, details);
+      }
+    });
+  }
+
+  onNewUtxo(callback: (utxo: UnblindedOutput) => Promise<void>) {
+    return Browser.storage.onChanged.addListener(async (changes, areaName) => {
+      if (areaName !== 'local') return;
+      const listUnspentKeys = Object.entries(changes).filter(
+        ([key, changes]) => ScriptUnspentsKey.is(key) && changes.newValue !== undefined
+      );
+
+      for (const [, change] of listUnspentKeys) {
+        // check if there is NEW utxo
+        const newUnspents = change.newValue as ListUnspentResponse | undefined;
+        if (!newUnspents) continue; // it means we just deleted the key
+        const oldUnspents = change.oldValue ? (change.oldValue as ListUnspentResponse) : [];
+
+        const oldUnspentsSet = new Set(oldUnspents);
+        const newUnspentsAdded = newUnspents.filter((unspent) => !oldUnspentsSet.has(unspent));
+        for (const newUtxo of newUnspentsAdded) {
+          const utxo = await this.getOutputBlindingData(newUtxo.tx_hash, newUtxo.tx_pos);
+          await callback(utxo);
+        }
+      }
+    });
+  }
+
+  onDeleteUtxo(callback: (utxo: UnblindedOutput) => Promise<void>): void {
+    return Browser.storage.onChanged.addListener(async (changes, areaName) => {
+      if (areaName !== 'local') return;
+      const listUnspentKeys = Object.entries(changes).filter(
+        ([key, changes]) => ScriptUnspentsKey.is(key) && changes.newValue !== undefined
+      );
+
+      for (const [, change] of listUnspentKeys) {
+        // check if there is NEW utxo
+        const newUnspents = change.newValue as ListUnspentResponse | undefined;
+        if (!newUnspents) continue; // it means we just deleted the key
+        const oldUnspents = change.oldValue ? (change.oldValue as ListUnspentResponse) : [];
+
+        const newUnspentsSet = new Set(newUnspents);
+        const oldUnspentsDeleted = oldUnspents.filter((unspent) => !newUnspentsSet.has(unspent));
+        for (const oldUtxo of oldUnspentsDeleted) {
+          const utxo = await this.getOutputBlindingData(oldUtxo.tx_hash, oldUtxo.tx_pos);
+          await callback(utxo);
+        }
+      }
+    });
+  }
+
+  private async getScripts(...networks: NetworkString[]): Promise<Array<string>> {
+    // get all scripts key in storage
+    const wholeStorage = await Browser.storage.local.get(null);
+    return Object.entries(wholeStorage)
+      .filter(
+        ([key, value]) =>
+          ScriptDetailsKey.is(key) && networks.includes((value as ScriptDetails).network)
+      )
+      .map(([key]) => ScriptDetailsKey.decode(key)[0]);
+  }
+
+  private async getAllAccountsNames(): Promise<Array<string>> {
+    const wholeStorage = await Browser.storage.local.get(null);
+    return Object.entries(wholeStorage)
+      .filter(([key, value]) => AccountKey.is(key) && value !== null)
+      .map(([key]) => AccountKey.decode(key)[0]);
+  }
+
+  static LOCKTIME = 5 * 60 * 1000; // 5 minutes
```

Comment:
Why do you lock outputs for 5 minutes?

Reply:
Instead of using an `until` timestamp, I just do this at coin selection:
```
const lockedOutpoints = selectedUtxos.map((utxo) => get().lockOutpoint(utxo));
await sleep(60_000);
lockedOutpoints.forEach((outpoints) => get().unlockOutpoint(outpoints));
```
Am I missing something?

Reply:
The main advantage to use a cached `until` timestamp is that if the process stops for any reason, we know that our outpoint will be unlocked anyway. This is perfect for the Marina context cause we don't know if the user will stop suddenly the extension/browser/popup.